CoursesManager.UI/ViewModels/Students/CourseDialogViewModel.cs

```diff
@@ -0,0 +1,226 @@
+                        Code = string.Empty,
+                        Description = string.Empty,
+                        Location = null,
+                        IsActive = false,
+                        StartDate = DateTime.Now,
+                        EndDate = DateTime.Now,
+                        Image = null,
+
+                    };
+
+            OriginalCourse = course;
+            Locations = new ObservableCollection<Location>(locationRepository.GetAll());
+
+            Courses = new ObservableCollection<string>(_courseRepository.GetAll().Select(c => c.Name));
+            SaveCommand = new RelayCommand(ExecuteSave, () => CanSave);
+            CancelCommand = new RelayCommand(OnCancel);
+            UploadCommand = new RelayCommand(UploadImage);
+
+            Course.PropertyChanged += (_, _) => UpdateCanSave();
+            UpdateCanSave();
+        }
+
+        private Course? OriginalCourse { get; }
+        protected override void InvokeResponseCallback(DialogResult<Course> dialogResult)
+        {
+            ResponseCallback.Invoke(dialogResult);
+        }
+
+        private void NotifyCanExecuteChanged()
+        {
+            (SaveCommand as RelayCommand)?.RaiseCanExecuteChanged();
+        }
+
+        private async void ExecuteSave()
+        {
+            try
+            {
+                await OnSaveAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+            }
+        }
+
+        private void UpdateCanSave()
+        {
+            var fieldsAreValid = !string.IsNullOrWhiteSpace(Course!.Name)
+                                 && !string.IsNullOrWhiteSpace(Course.Code)
+                                 && Course.StartDate != default
+                                 && Course.EndDate != default
+                                 && Course.Location != null
+                                 && !string.IsNullOrWhiteSpace(Course.Description);
+
+            Console.WriteLine($"Fields Valid: {fieldsAreValid}");
+
+            var noValidationErrors = !HasValidationErrors();
+
+            Console.WriteLine($"No Validation Errors: {noValidationErrors}");
+
+            CanSave = fieldsAreValid && noValidationErrors;
+
+            NotifyCanExecuteChanged();
+        }
+
+
+        private bool HasValidationErrors()
+        {
+            if (Application.Current.MainWindow == null)
+            {
+                return false;
+            }
+
+            foreach (var child in LogicalTreeHelper.GetChildren(Application.Current.MainWindow))
+            {
+                if (child is DependencyObject dependencyObject && Validation.GetHasError(dependencyObject))
+                {
+
+                    if (dependencyObject is Image && Course!.Image == null)
+                        continue;
+
+                    return true;
+                }
+            }
+            return false;
+        }
```

Comment:
Dit is niet echt wenselijk denk ik. Waar liep je tegenaan dat dit nodig was? In principe is het niet de bedoeling dat je window gerelateerde dingen doet in de viewmodel. De viewmodels mogen eigenlijk niks weten van de view en moeten compleet los staan van elkaar.